src/core/util/byte_stream.cc

```diff
@@ -100,6 +100,8 @@ void OutputByteStream::ProduceData(std::size_t amount) {
 }
 
 void OutputByteStream::WriteData(const std::uint8_t* data, std::size_t len) {
+  if (!len)
+    return;
```

Comment:
>IMHO, the packet builder should not care about these kind of rules. They should only be expressed in session.cc.

Sure, but the code you've changed is generic byte stream code that can (and will) be applied throughout the codebase - and is not specific to SSU. If you want SSU specificity, why not work on the packet builder specifically?

>So what do you think about debug log if data null and len null, otherwise error log + throw exception ?

This sounds like an anti-pattern in the making. If we write data and expect said data to *not* fulfill those parameters but said so data so happens *to* fulfill those parameters, we'll probably spend excessive time debugging instead of implementing any needed design changes. Can you think of any other use-case for this byte stream diff outside of SSU?